src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -443,10 +443,56 @@ object Parser {
           val pres = clauses.collect{ case x: RequiresClause => x.exp }
           val preserves = clauses.collect{ case x: PreservesClause => x.exp }
           val posts = clauses.collect{ case x: EnsuresClause => x.exp }
+          val terminationMeasure = {
+            val t = clauses.collect{ case x: DecreasesClause => x.measure}
+            if(t.size <= 1){
+              t.headOption
+            } else {
+              Violation.violation("Unexpected amount of decreases clause")
+            }
+          }
           val isPure = pure.nonEmpty || clauses.contains(PureClause)
-          PFunctionSpec(pres, preserves, posts, isPure)
+          PFunctionSpec(pres, preserves, posts, terminationMeasure, isPure)
       }
     }
+    
+    lazy val measures:Parser[PTerminationMeasure]=
+    "*"  ^^^ PStarCharacter() | ("_" <~ eos) ^^^ PUnderscoreCharacter() | repsep(expression,",") <~ eos ^^ PTupleTerminationMeasure | firstConditionalMeasure 
+
+    lazy val firstConditionalMeasure: Parser[PConditionalMeasureCollection]=
+      repsep(expression,",") ~ ("if" ~> expression <~ eos ) ~ ConditionalUnderscore.? ^^ {
+        case expression ~ condition ~ Some(PConditionalMeasureCollection(tuple)) => PConditionalMeasureCollection(Vector(PConditionalMeasureExpression(expression, condition)) ++ tuple)
+        case expression ~ condition ~ None => PConditionalMeasureCollection(Vector(PConditionalMeasureExpression(expression, condition)))
+      } | (repsep(expression,",") <~ eos) ~ ConditionalUnderscore.? ^^ {
+        case expression ~ Some(PConditionalMeasureCollection(tuple)) => PConditionalMeasureCollection(Vector(PConditionalMeasureExpression(expression, PBoolLit(true))) ++ tuple)
+        case expression ~ None => PConditionalMeasureCollection(Vector(PConditionalMeasureExpression(expression, PBoolLit(true))))
+      } | "_"  ~  ("if" ~> expression <~ eos ) ~ ConditionalExpressions.? ^^ {
+        case _ ~ condition ~ Some(PConditionalMeasureCollection(tuple)) => PConditionalMeasureCollection(Vector(PConditionalMeasureUnderscore(condition)) ++ tuple)
+        case _ ~ condition ~ None => PConditionalMeasureCollection(Vector(PConditionalMeasureUnderscore(condition)))
+      } | ("_" <~ eos) ~ ConditionalExpressions.? ^^ {
+        case _ ~ Some(PConditionalMeasureCollection(tuple)) => PConditionalMeasureCollection(Vector(PConditionalMeasureUnderscore(PBoolLit(true))) ++ tuple)
+        case _ ~ None => PConditionalMeasureCollection(Vector(PConditionalMeasureUnderscore(PBoolLit(true))))
+      }
+
+    lazy val ConditionalUnderscore: Parser[PConditionalMeasureCollection]=
```

Comment:
The name of this parser is misleading. In particular, it suggests that it parses measures in the form `decreases _ if [ measure ]`. However, it also parses `decreases *`.